core/editable.js

```diff
@@ -2556,7 +2557,7 @@
 		};
 	} )();
 
-	function mergeBlocksCollapsedSelection( editor, range, backspace, startPath ) {
+	function mergeBlocksCollapsedSelection( editor, range, backspace, startPath, skipRangeTrimming ) {
```

Comment:
By https://github.com/ckeditor/ckeditor4/pull/4746#discussion_r650770440 I meant removing this `skipRangeTrimming` parameter from function signature and instead either just pass `true` later:
```js
if ( !range[ backspace ? 'checkStartOfBlock' : 'checkEndOfBlock' ]( true ) )
```

or initialise the boolean variable inside this function:
```js
var skipRangeTrimming = true;

if ( !range[ backspace ? 'checkStartOfBlock' : 'checkEndOfBlock' ]( skipRangeTrimming ) )
```

Reply:
That is true, but I want to be 100% sure, that this boolean comes from a callback attached to `key` event. Right now, that is the only use case, but if someone ever thinks about usage `mergeBlocksCollapsedSelection` will not affect this logic.